src/SettingsMigrator/Migrator.cs

```diff
@@ -241,11 +249,12 @@ private void RemoveEmbeddedTestSettings(XmlDocument newXmlDoc)
         /// <param name="testSettingsNodes">testSettingsNodes</param>
         /// <param name="testTimeout">testTimeout</param>
         /// <param name="parallelTestCount">parallelTestCount</param>
+        /// <param name="hostProcessPlatform">hostProcessPlatform</param>
         /// <param name="newXmlDoc">newXmlDoc</param>
-        private void AddLegacyNodes(TestSettingsNodes testSettingsNodes, string testTimeout, string parallelTestCount, XmlDocument newXmlDoc)
+        private void AddLegacyNodes(TestSettingsNodes testSettingsNodes, string testTimeout, string parallelTestCount, string hostProcessPlatform, XmlDocument newXmlDoc)
         {
-            if (!(testSettingsNodes.Deployment != null || testSettingsNodes.Script != null || testSettingsNodes.UnitTestConfig != null ||
-                !string.IsNullOrEmpty(parallelTestCount) || !string.IsNullOrEmpty(testTimeout) || testSettingsNodes.Hosts != null))
+            if (testSettingsNodes.Deployment == null && testSettingsNodes.Script == null && testSettingsNodes.UnitTestConfig == null &&
```

Comment:
Feel this check is redundant. Let's discuss.

Reply:
This check is needed because in case none of these settings is present we want to avoid creating a redundant Legacy node and a forced legacy node